File: LATIHAN4DPBO2023/CPP/Human.cpp
/*Saya Mochamad Khaairi NIM 2106416 mengerjakan Latihan 4
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan 
seperti yang telah dispesifikasikan. Aamiin.*/

//include file header
#include "header.hh"

//inisialisasi constructor
Human::Human()
{
	nik = "";
	name = "";
	gender = "";
}

//inisialisasi constructor dengan parameter
Human::Human(string nik, string name, string gender)
{
	this->nik = nik;
	this->name = name;
	this->gender = gender;
}

//inisiliasasi method setter dan getter
void Human::setNik(string nik)
{
	this->nik = nik;
}

string Human::getNik()
{
	return this->nik;
}

void Human::setName(string name)
{
	this->name = name;
}

string Human::getName()
{
	return this->name;
}


void Human::setGender(string gender)
{
	this->gender = gender;
}

string Human::getGender()
{
	return this->gender;
}

//inisialisasi destructor
Human::~Human()
{
}

File: LATIHAN4DPBO2023/Python/Human.py
#inisialisasi class Human
class Human:
	# atribut
	__nik = ""
	__name = ""
	__gender = ""

	# constructor
	def __init__(self, nik = "", name = "", gender = ""):
		self.__nik = nik
		self.__name = name
		self.__gender = gender

	# kumpulan method setter dan getter untuk tiap atribut
	def setNik(self, nik):
		self.__nik = nik

	def getNik(self):
		return self.__nik

	def setName(self, name):
		self.__name = name

	def getName(self):
		return self.__name

	def setGender(self, gender):
		self.__gender = gender

	def getGender(self):
		return self.__gender